### PR TITLE
edits typedefs to include exercises and exerciseCrud

### PR DESCRIPTION
**Pull request by mino323**
This Pr edits the graphql typedef to include exercise and exercise Crud. 
I also ran ```yarn generate``` to update index.tsx
Exercises will be the foundation to the c0d3 dojo.

### CODE REVIEW

File: graphql/typeDefs.ts [hunk 1]
```diff
@@ -57,6 +58,22 @@ export default gql`
     deleteComment(id: Int!): Comment
     addModule(lessonId: Int!, name: String!, content: String!): Module
     deleteModule(id: Int!): SuccessResponse
+    addExercise(
+      moduleId: Int!
+      description: String!
+      answer: String!
+      testable: Boolean!
+      testStr: String
```
Comment (anthonykhoa) [inline graphql/typeDefs.ts:66]: What is this field used for? `testStr`

Comment (songz) [reply graphql/typeDefs.ts:66]: if user wants to add a test to be run:

```js
test('should equal 55', () => {
  expect(inputValue).toEqual(55)
}
```

Comment (anthonykhoa) [reply graphql/typeDefs.ts:66]: Thanks!

File: graphql/typeDefs.ts [hunk 2]
```diff
@@ -57,6 +58,22 @@ export default gql`
...
+    ): Exercise
```
Comment (anthonykhoa) [inline graphql/typeDefs.ts:67]: @mino323 Question. Why is it for `addExercise` we return type `Exercise`, but for ` updateExercise`, we return `Exercise!`?(One uses exclamation point, the other doesn't).

**Comment by anthonykhoa** [comment]
Nice PR description

File: graphql/typeDefs.ts (hunk 2 above)
Comment (mino323) [reply graphql/typeDefs.ts:67]: My thinking was when you add an exercise you might not need anything returned. When you update you want to see the updated data.

Comment (SahilKalra98) [reply graphql/typeDefs.ts:67]: My thinking is whenever you create/update/delete a resource, it should return that resource. Not sure though

Comment (flacial) [reply graphql/typeDefs.ts:67]: I agree with Karla. In my point of view, it works as a confirmation that this request is successful.

Comment (songz) [reply graphql/typeDefs.ts:67]: @SahilKalra98 is correct. whenever you create/update a resource, unless there is error you should be returning the resource. I'm not sure about delete though, my memory tells me you simply return the id of the deleted resource (no point returning a data that no longer exists).

Comment (mino323) [reply graphql/typeDefs.ts:67]: Got it thanks, all!

**Review by flacial — APPROVED** [review]

**Pull request merged**